netsim/modules/ripv2.py

```diff
@@ -5,8 +5,33 @@
 from box import Box
 
 from ..augment import devices
+from ..utils import log
 from . import _Module, _routing, bfd
 
+"""
+ripv2_unnumbered: Check whether RIPv2 is used on unnumbered interfaces and verify
+device support via the ripv2.unnumbered feature flag.
+
+An interface is unnumbered if it has IPv4 enabled but no IPv4 address (ipv4: true)
+"""
+def ripv2_unnumbered(node: Box, features: Box) -> bool:
+  is_unnumbered = False
+
+  for intf in node.get('interfaces',[]):
+    if intf.get('ipv4',None) is True and 'ripv2' in intf:
+      is_unnumbered = True
+      break
+
+  if is_unnumbered:
+    if not features.ripv2.get('unnumbered',False):
+      log.error(
```

Comment:
It would be nice to collect the offending interface and link names and report them in the error message